examples/complete/main.tf

```diff
@@ -53,6 +53,21 @@ module "cloudfront" {
     }
   }
 
+  create_vpc_origin = true
+  vpc_origin = {
+    ec2_vpc_origin = {
+      name                   = local.subdomain
+      arn                    = module.ec2.arn
```

Comment:
Can we have a fake `arn` here and be able to run "terraform apply"? If so, let's remove `module.ec2.arn`; and eliminate unnecessary EC2 and VPC resources.

Reply:
Hi Anton, thanks for the review.

I understand the EC2 and VPC resources could introduce too much noise but they are fundamental to make `examples/complete` to work and continue being a meaningful example.

I tried to input a fake EC2 ARN but it fails because it expects a valid one while creating the resource:
```
│ Error: creating CloudFront VPC Origin
│
│   with module.cloudfront.aws_cloudfront_vpc_origin.this["ec2_vpc_origin"],
│   on ../../main.tf line 28, in resource "aws_cloudfront_vpc_origin" "this":
│   28: resource "aws_cloudfront_vpc_origin" "this" {
│
│ operation error CloudFront: CreateVpcOrigin, https response error StatusCode: 400, RequestID: 4cbe2b60-a6f1-45d6-a04c-1498de54e25c, InvalidArgument: The
│ parameter VpcOriginEndpointConfig Arn Does Not Exist.
```

Regarding the remaining resources, let me detail why each of them is important:
- VPC:
  - The EC2 instance runs on top of it
  - The CloudFront VPC Origin creates an ENI
- VPC Endpoint (S3 Gateway)
  - The chepeast solution to let the EC2 instance to access AL2023's repository and install NGINX ([link to explanation](https://repost.aws/knowledge-center/ec2-al1-al2-update-yum-without-internet))
- EC2 Security Group:
  - Required by the EC2 by default
  - The ingress rule allows the VPC Origin's security group to access the NGINX service listening on TCP/80
  - The egress rule allows connection to AL2023's repository on S3 and VPC Origin egress.
- The `aws_security_group.vpc_origin` queries the VPC Origin's security group ID --that is created by AWS automatically-- that's needs to be added into the EC2's security group
- The remaining data resources are just helpers.
- NGINX installation: I considered nice having the cache behavior `/vpc-origin/*` responding with a simple web server response instead of a CloudFront error.

You can find a detailed list of requirements [here](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-vpc-origins.html#vpc-origin-prerequisites).

I realised that my EC2 example included an unneeded IAM role that I forgot to remove (I was troubleshooting why NGINX was not installed properly). I will remove it in my next commit.

Do you think we are OK keeping all these resources?

Reply:
I have simplified the example to just make an EC2 instance, leaving all sort of customization to the user (availability, security, cost, best practices, etc - is always left for the users).